share/config.fish

```diff
@@ -195,6 +195,68 @@ function . --description 'Evaluate contents of file (deprecated, see "source")'
 	end
 end
 
+# Try to set the locale from the system configuration if we did not inherit any.
```

Comment:
I don't buy that argument. There are plenty of things in private fish functions that would cause problems if people randomly ran them. The whole point of the `__fish_*` naming convention is to clearly signify the contents are private to fish and should be treated as opaque blobs. Anyone running one of those scripts by hands shouldn't be surprised if the result is the equivalent of running the "HCF" machine opcode (halt, catch fire). One of the things that drives me a little crazy are all the functions, both C++ and fish script, that are uncomfortably large. Too, in this case running it by hand should be harmless since the first thing you do is test whether any locale var has been set and if true exit without changing anything.